### PR TITLE
fix(compose): refresh unseeded worker RNG from PyTorch DataLoader

### LINKED ISSUE #266: [Bug] Workers produce identical augmentations per batch position across epochs when used with PyTorch DataLoader
### Environment:

albucore                0.0.24
albumentations          2.0.8
opencv-python-headless  4.13.0.92
torch: 2.11.0
Python: 3.14.4
OS: CachyOS

### Description

When using A.Compose inside a PyTorch Dataset.__getitem__ with num_workers > 0, each sample at a given batch position receives identical augmentations across epochs. The image content varies correctly (shuffle is working) but the augmentation applied — including randomized parameters like dropout position, shear amount, flip decisions — is identical for every image processed at the same intra-batch position.

The issue is resolved by setting num_workers=0.

Minimal reproducible example

```
import albumentations as A
import numpy as np
import torch
from torch.utils.data import Dataset, DataLoader
import matplotlib.pyplot as plt
import cv2
cv2.setNumThreads(0)

class TestDataset(Dataset):
    def __init__(self):
        self.augmentations = A.Compose([
            A.HorizontalFlip(p=0.5),
            A.Affine(p=0.5),
            A.CoarseDropout(num_holes_range=(1,1), p=0.5)
        ])
        self.data = np.ones((100, 32, 32, 3), dtype=np.uint8)*255

    def __len__(self):
        return len(self.data)

    def __getitem__(self, idx):
        img = self.augmentations(image=self.data[idx])['image']
        return torch.from_numpy(img).permute(2,0,1).float() / 255.0

if __name__ == "__main__":
    ds = TestDataset()
    dl = DataLoader(ds, batch_size=16, shuffle=True, num_workers=4)
    
    for runs in range(2):
        # Run two epochs and compare augmentations at position 0
        batches = []
        for epoch in range(2):
            for batch in dl:
                batches.append(batch)
                break
        
        print("Identical augmentation at position 0 across epochs:", 
              (batches[0][0] == batches[1][0]).all().item())
        
        plt.figure()
        plt.imshow(batches[0][0].permute(1,2,0))
        plt.show()
        plt.figure()
        plt.imshow(batches[1][0].permute(1,2,0))
        plt.show()
```

### PR DESCRIPTION
**Pull request by federicopozzi33**
Before the fix, worker reseeding exited early when `self._base_seed` was `None`. That meant pipelines created without an explicit `seed=` did not synchronize their random state from the PyTorch worker seed inside worker processes. As a result, the _Compose_ instance kept using the same internal RNG state pattern for a given worker/process position, which is why the same batch position could receive the same augmentation pattern across epochs.

Closes: #266

### CODE REVIEW

**Comment by ternaus** [comment]
@federicopozzi33 Thanks!
